packages/components/src/Chip/components/ChipSuffix/Chip.Suffix.tsx

```diff
@@ -4,28 +4,58 @@ import { Icon } from "@jobber/components/Icon";
 import { useChildComponent } from "../../hooks";
 import styles from "../../Chip.css";
 
-export function ChipSuffix({ children, className }: ChipSuffixProps) {
+export function ChipSuffix({
+  children,
+  className,
+  onClick,
+  testID,
+  ariaLabel,
+}: ChipSuffixProps) {
   let singleChild = useChildComponent(children, d => d.type === Icon);
 
+  const handleClick = (
+    event:
+      | React.MouseEvent<HTMLSpanElement, MouseEvent>
+      | React.KeyboardEvent<HTMLSpanElement>,
+  ) => {
+    if (!onClick) {
+      return;
+    }
+    event.preventDefault();
+    event.stopPropagation();
+    onClick(event);
+  };
+
   if (!allowedSuffixIcons.includes(singleChild?.props?.name)) {
     singleChild = undefined;
   }
 
-  return (
-    <span
-      className={classNames(
-        styles.suffix,
-        className,
-        !singleChild && styles.empty,
-      )}
-    >
-      {singleChild}
-    </span>
-  );
+  const tagProps = {
+    className: classNames(
+      onClick ? styles.clickableSuffix : styles.suffix,
+      className,
+      !singleChild && styles.empty,
+    ),
+    onClick: handleClick,
+    onKeyPress: handleClick,
+    ...(onClick ? { role: "button" } : {}),
+    ...(onClick ? { tabIndex: 0 } : {}),
+    "data-testid": testID,
+    "aria-label": ariaLabel,
+  };
+
+  return <span {...tagProps}>{singleChild}</span>;
```

Comment:
Out of curiosity and just to educate myself, is this how you typically define component props? I.e.
```jsx
const tagProps = {...};
 
<span {...tagProps}>{singleChild}</span>;
```
rather than
```jsx
<span
  className={classNames(
    onClick ? styles.clickableSuffix : styles.suffix,
    className,
    !singleChild && styles.empty,
  )}
  onClick={handleClick}
  onKeyPress={handleClick}
  role={onClick ? "button" : undefined}
  tabIndex={onClick ? 0 : undefined}
  data-testid={testID}
  aria-label={ariaLabel}
>
  {singleChild}
</span>
```

I'm always trying to be cautious with prop spreading, since there's a risk of unintentionally passing unsupported props. ESLint even has [a rule for this](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-props-no-spreading.md).

Reply:
Great question. So, when I was initially using an `if` statement - spreading the props would have been our go-to pattern.   You can see that pattern being used in Button [here](https://github.com/GetJobber/atlantis/blob/master/packages/components/src/Button/Button.tsx#L174).  

But now that we're avoiding that `if` statement to determine the use of `span` or `button`, we can put the props back to the more standard composition.  I've made that change here 9ab340194198399a34a0ef06bfb83feaa6582da8